packages/react-query/src/internals/context.tsx

```diff
@@ -94,19 +112,19 @@ export interface TRPCContextState<
   /**
    * @link https://tanstack.com/query/v4/docs/reference/QueryClient#queryclientfetchquery
    */
-  fetchQuery<
+  fetchQuery: <
```

Comment:
https://typescript-eslint.io/rules/unbound-method

Marking these as properties with function types (rather than methods) indicates it's safe to call them without caring about the `this` scope.


```ts
declare const values: {
  a(): void;
  b: () => void;
}

const { a, b } = values;

a(); // Not safe! What if the `this` is wrong?

b(); // this is fine
```

I actually filed https://github.com/typescript-eslint/typescript-eslint/issues/6276 from working on these changes.

Reply:
FWIW, I really don't care about this one as we're moving away form using `class`es altogether b/c JS sucks at them

Reply:
Nice. In that case you might want https://typescript-eslint.io/rules/method-signature-style, to enforce folks don't accidentally imply they're creating methods (& educate on the difference).